src/DetailsView/reports/components/outcome-summary-bar.tsx

```diff
@@ -1,33 +1,37 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { kebabCase } from 'lodash';
 import * as React from 'react';
 
 import { NamedSFC } from '../../../common/react/named-sfc';
-import { OutcomeIcon } from './outcome-icon';
-import { allRequirementOutcomeTypes, RequirementOutcomeStats } from './requirement-outcome-type';
+import { outcomeIconMap, outcomeIconMapInverted, OutcomeStats, OutcomeType, outcomeTypeSemantics, OutcomeUnits } from './outcome-type';
 
-type OutcomeUnits = 'percentage' | 'requirements';
-
-const outcomeText = {
-    pass: 'Passed',
-    incomplete: 'Incomplete',
-    fail: 'Failed',
+export type OutcomeSummaryBarProps = {
+    outcomeStats: Partial<OutcomeStats>;
+    allOutcomeTypes: OutcomeType[];
+    units?: OutcomeUnits;
+    inverted?: boolean;
 };
 
-export type OutcomeSummaryBarProps = RequirementOutcomeStats & { units?: OutcomeUnits };
+export const OutcomeSummaryBar = NamedSFC<OutcomeSummaryBarProps>('OutcomeSummaryBar', props => {
+    return (
+        <div className="outcome-summary-bar">
+            {props.allOutcomeTypes.map(outcomeType => {
+                const { units, inverted } = props;
+                const text = outcomeTypeSemantics[outcomeType].pastTense;
+                const iconMap = inverted === true ? outcomeIconMapInverted : outcomeIconMap;
+                const outcomeIcon = iconMap[outcomeType];
+                const count = props.outcomeStats[outcomeType];
+                const suffix = units === 'percentage' ? '% ' : '';
```

Comment:
It seems we can promote the `suffix` to be a prop instead of having this tertiary operator here, that way, if we need a different, new suffix, it'll be up the the user class to pass the proper suffix (this align with open/close principle by the way).
For the name of the property, maybe `countSuffix1` is good enough